src/encoder.rs

```diff
@@ -2424,27 +2414,40 @@ fn encode_partition_bottomup(
                     offset,
                     pmvs,//&best_decision.mvs[0]
                     best_rd
-                ) {
-                    rd_cost += cost;
-                    if rd_cost >= best_rd || rd_cost >= ref_rd_cost {
-                        assert!(cost != std::f64::MAX);
-                        early_exit = true;
-                        if partition == PartitionType::PARTITION_SPLIT { break; }
+                );
+                let cost = rdo_output.rd_cost;
+                assert!(cost >= 0.0);
+                if cost == std::f64::MAX { continue; }
+
+                rd_cost += cost;
+                if rd_cost >= best_rd || rd_cost >= ref_rd_cost {
+                    assert!(cost != std::f64::MAX);
+                    early_exit = true;
+                    if partition == PartitionType::PARTITION_SPLIT { break; }
+                }
+                else {
+                    if partition != PartitionType::PARTITION_SPLIT {
+                        child_modes.push(rdo_output.part_modes[0].clone());
                     }
-                    else { child_modes.push(mode_decision); }
                 }
             };
 
             if !early_exit && rd_cost < best_rd {
                 best_rd = rd_cost;
                 best_partition = partition;
-                best_pred_modes = child_modes.clone();
+                if partition != PartitionType::PARTITION_SPLIT {
+                    assert!(child_modes.len() > 0);
+                    rdo_output.part_modes = child_modes.clone();
```

Comment:
`clone()` seems unnecessary, doesn't `child_modes` goes out of scope immediately below?

Reply:
Thanks for catching this, I was also wanting check which clone() I can remove, since it increases the encoding time. I will try again with removing it. I've only added clone() to please rustc. :)

Reply:
You were right, that was the last use of it. So I can remove it.